### PR TITLE
Airtable refactor, bug fixes, Auth overhaul

### LINKED ISSUE #69: Change `Login` and `Logout` to `Log In` and `Log Out`
For consistency with `Sign Up` / `SignUpScreen`, update both copy and variable names to be two words for `Login` and `LogOut`

### PR DESCRIPTION
**Pull request by anniero98**
[//]: # "These comments are meant for your reference. They are invisible and don't need to be deleted!"

# What's new in this PR

Refactoring our Airtable code in this repo to use `airtable-schema-generator` (now that it's been updated to 1.3.3 with some nice features!).

### REQUIRES `.env.development`/`production` to be changed from `AIRTABLE_API_KEY` to `REACT_APP_AIRTABLE_API_KEY`
...for compatibility for v1.3.3

Has the same Airtable schme updates as https://github.com/calblueprint/dccentralkitchen-clerks/pull/29; Points History screen will not load without transaction columns being changed to `Total Sale`.

- Removed code in `common.js` since `BASE` will no longer be used

### Map
- Bug: initial store showed the incorrect products
- Edge case: If no products are linked to a store, the key `productIds` will not exist on the `store` object, so special handling is required

### Navigation
- Cleanup: refactored `AppNavigator` by extracting `DrawerContent` as a class, and moving `AuthStack` to `StackNavigators`

### News
- Bug: `NewsScreen` displayed `Created` date, not `Post Date`

### Rewards
- Deprecated: removed pull-to-refresh implementation to declutter the file. We can add it back from a different trigger if necessary. Removed deprecated props which were not being used in `RewardsHome` and `PointsHistory` components.
- Cleanup: Made `Overline` components normally-cased (since the styled component uses `text-transform`)
- Cleanup: Added `isLoading` to `DrawerContent` and `RewardsScreen`. This is probably the most crucial place to add a loading screen (@ace) imo; it's noticeably laggy here.
- Cleanup: Add `Rewards.js` to `constants` so things aren't hardcoded.

### Auth
Integrated @tommypoa 's changes from `material-ui` branch

### SignUp
- Fixed SignUp's duplicate customer check to actually work, but no design exists for displaying the error right now (@12aschen)
- Replaced the `name, nameError` etc in `this.state` to use an object similar to `indicators`.
- Changed behavior of `handleErrorState` to update error messages from `validate.js` instead of being hardcoded into `AuthField`
- Added an error message and error state for `name`
- Made `onTextChange` hooks async so that error handling would happen _after_ values updated in state; and handled permissions
* Note that `onBlur` is still required for the edge case of user tapping in then tapping out
- Added margin to the SignUp button

### Login
- Added error message and no error message but error indicators for that state.
- Changed `Login` to `LogIn`/`Log In` where necessary (and `Logout` in the drawer)
- closes #69
- Hardcoded the size of `TextFieldContainer` (losing transition animations to error states) to do so.

[//]: # "Describe what's new in this PR in a few lines. A description and bullet points for specifics will suffice."

### Related PRs

[//]: # "Optional - related PRs you're waiting on/ PRs that will conflict, etc; if this is a refactor, feel free to add PRs that previously modified this code"

Waiting on @tommypoa for some last updates to auth 

## How to review

Nothing should really change appearance-wise or functionality-wise for anything except the Signup/Login flow! But everything should still work (^:

[//]: # "The order in which to review files and what to expect when testing locally"

## Next steps
Would be nice to change `userId` to `customerId` for consistency between the repos.

[//]: # "What's NOT in this PR, doesn't work yet, and/or still needs to be done"

## Tests Performed, Edge Cases

[//]: # "Hopefully we will add a testing suite/CI soon, but until then note down the steps you took to test locally"

Signup: https://www.loom.com/share/896a8bb1e5ba442281908122fc35e298

Login: https://www.loom.com/share/947060a3784242bc8e79f91b24c8c386

### Screenshots

[//]: # "Add screenshots of expected behavior - GIFs if you're feeling fancy!"

CC: @wangannie

[//]: # "This tags in both Annies as a default. Feel free to change, or add on anyone who you should be in on the conversation."

### CODE REVIEW

File: prettier.config.js [hunk 1]
```diff
@@ -1,5 +1,5 @@
 module.exports = {
-  printWidth: 80,
+  printWidth: 120,
```
Comment (anniero98) [inline prettier.config.js]: This was a mistake 🙃 No wonder the diffs have been so big lmao I'll fix tomorrow

File: .env.example [hunk 2]
```diff
@@ -1,3 +1,7 @@
-AIRTABLE_BASE_ID='your-API-key-here'
-AIRTABLE_API_KEY='keyioENo5dwh9Czt9'
+REACT_APP_AIRTABLE_API_KEY='AIRTABLE API KEY'
```
Comment (wangannie) [inline .env.example:1]: Can we have this formatting match Clerk with the comments and everything?

File: navigation/DrawerContent.js [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+          }}>
+          <TouchableOpacity
+            style={{ padding: 16 }}
+            onPress={() => Linking.openURL(this.state.link)}>
+            <Title>Report Issue</Title>
+          </TouchableOpacity>
+          <TouchableOpacity
+            style={{ paddingLeft: 16, paddingBottom: 21 }}
+            onPress={() => this._logout()}>
+            <Title>Logout</Title>
```
Comment (wangannie) [inline navigation/DrawerContent.js:71]: Isn't this supposed to be 2 words now

File: components/store/StoreCard.js [hunk 4]
```diff
@@ -67,17 +76,17 @@ export default function StoreCard({ store, callBack, seeProduct }) {
         {seeProduct && <StoreProductButton callBack={callBack} />}
       </SpaceBetweenRowContainer>
       <Caption style={{ marginBottom: 4 }} color={Colors.secondaryText}>
-        {distance} miles away
+        {`${distance} miles away`}
```
Comment (wangannie) [inline components/store/StoreCard.js:79]: Not super related but now I'm thinking we might need to look into a different way to calculate distance. Currently, it's always unrealistically low bc it's calculated as the crow flies.

File: lib/mapUtils.js [hunk 5]
```diff
@@ -1,74 +1,36 @@
...
-        resolve(stores);
-      })
-      .catch(err => reject(err));
-  });
-};
+export async function getStoreData() {
+  const records = await getAllStores();
+  // Filter out the Clerk Training store
+  const stores = records
+    .filter(record => record.id !== 'recq6630DixVw63un')
```
Comment (wangannie) [inline lib/mapUtils.js:19]: Can we (/ should we) also filter out the dev clerk training? `reck65q6wsvyQvnLf`

File: navigation/StackNavigators.js [hunk 6]
```diff
@@ -16,6 +23,15 @@ const config = Platform.select({
   },
 });
 
+export const AuthStack = createStackNavigator(
+  {
+    Welcome: WelcomeScreen,
+    Login: LoginScreen,
```
Comment (wangannie) [inline navigation/StackNavigators.js:29]: Since we're making both Log In and Sign Up 2 words across the board, would it make sense to have it be `LogIn` everywhere to match `SignUp`

File: screens/auth/SignUpScreen.js [hunk 7]
```diff
@@ -50,34 +50,44 @@ export default class SignUp extends React.Component {
...
-  _asyncSignin = async () => {
-    await AsyncStorage.setItem('userId', this.state.id);
+  _asyncSignin = async customerId => {
+    await AsyncStorage.setItem('userId', customerId);
     this.props.navigation.navigate('App');
   };
 
   registerForPushNotificationsAsync = async () => {
-    if (Constants.isDevice) {
+    if (Constants.isDescvice) {
```
Comment (wangannie) [inline screens/auth/SignUpScreen.js:90]: Typo?

File: screens/auth/SignUpScreen.js [hunk 8]
```diff
@@ -92,193 +102,142 @@ export default class SignUp extends React.Component {
...
+      try {
+        // Check for duplicates first
+        const formattedPhoneNumber = formatPhoneNumber(
+          this.state.values[signUpFields.PHONENUM]
+        );
+        const duplicateCustomers = await getCustomersByPhoneNumber(
+          formattedPhoneNumber
+        );
+        if (duplicateCustomers.length !== 0) {
+          console.log('Duplicate customer');
```
Comment (wangannie) [inline screens/auth/SignUpScreen.js:171]: It still let me create duplicate customers if I just spam click the button. I could definitely see this being an issue since it takes a while to see any response and there's no loading screen...
image

File: screens/auth/LoginScreen.js [hunk 9]
```diff
@@ -80,102 +75,102 @@ export default class Login extends React.Component {
...
+        const customer = customers[0];
+        // If customer exists, we should update their push tokens
+        await updateCustomerPushTokens(customer, token);
+        // Log in
+        await this._asyncLogin(customer.id);
+      } else if (customers.length > 1) {
+        // In case of database malformation, may return more than one record
+        // TODO this message is a design edge case
+        error =
+          'Database error: more than one customer found with this login information. Please report an issue so we can fix it for you!';
```
Comment (wangannie) [inline screens/auth/LoginScreen.js:111]: Well the 'report issue' link only shows in the hamburger menu which requires you to be logged in 😅 @12aschen

File: .env.example [hunk 10]
```diff
@@ -1,3 +1,7 @@
-AIRTABLE_BASE_ID='your-API-key-here'
-AIRTABLE_API_KEY='keyioENo5dwh9Czt9'
```
[the remark this answers is not in the document]
Comment (anniero98) [reply .env.example:2]: Github is probably picking up the old diff. It's the old key

File: components/rewards/RewardsHome.js [hunk 11]
```diff
@@ -39,20 +42,20 @@ function RewardsHome({ user }) {
...
           }}
-          progress={(parseInt(user.points) % 1000) / 1000}
+          progress={pointsToNext / rewardPointValue}
           color={Colors.primaryGreen}
         />
         <Body style={{ marginBottom: 28 }}>
-          Earn {`${1000 - (parseInt(user.points) % 1000)}`} points to unlock
-          your next $5 reward.
+          {`Earn ${rewardPointValue -
+            pointsToNext} points to unlock your next $5 reward`}
```
[the remark this answers is not in the document]
Comment (anniero98) [reply components/rewards/RewardsHome.js:50]: oops didn't realize it was being used anywhere

File: components/store/StoreCard.js (hunk 4 above)
Comment (anniero98) [reply components/store/StoreCard.js:79]: Conversation in #68

File: constants/Rewards.js [hunk 12]
```diff
@@ -0,0 +1 @@
+export const rewardPointValue = 1500;
```
[the remark this answers is not in the document]
Comment (anniero98) [reply constants/Rewards.js:1]: Added it now, I missed the instances where $5 was displayed while refactoring

File: navigation/StackNavigators.js (hunk 6 above)
Comment (anniero98) [reply navigation/StackNavigators.js:29]: yup sgtm

File: navigation/DrawerContent.js (hunk 3 above)
Comment (anniero98) [reply navigation/DrawerContent.js:71]: I guess so? LOL

File: lib/mapUtils.js (hunk 5 above)
Comment (anniero98) [reply lib/mapUtils.js:19]: Yup it's probably best to filter both since it's not breaking to do so, and we won't have to update code when we push to PROD

File: screens/auth/SignUpScreen.js (hunk 8 above)
Comment (anniero98) [reply screens/auth/SignUpScreen.js]: Fixed this by adding a `processing` variable in state.

**Comment by anniero98** [comment]
Realized that we shouldn't have been keeping permissions in state at all, and the unnecessary async/await pattern was making things messy/slow, so updated that ;; eliminated some code in `SignUp` and `LogIn` (':

**Comment by anniero98** [comment]
https://www.loom.com/share/de3bf3929ace40c1a86b712e1fdf5c62?focus_title=1&muted=1 last updates

File: lib/authUtils.js [hunk 13]
```diff
@@ -1,174 +1,98 @@
...
-      .catch(err => reject(err));
-  });
+  password: {
+    presence: {
+      message: '^Password cannot be blank.',
+    },
+    length: {
+      minimum: 8,
+      maximum: 20,
+      message: '^Must be 8-20 characters long',
```
[the remark this answers is not in the document]
Comment (anniero98) [reply lib/authUtils.js:66]: it's being checked fwiw with the 'maximum: 20' line
